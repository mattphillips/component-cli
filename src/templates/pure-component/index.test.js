import each from 'jest-each';
import pureComponent from './';

describe('Pure Component', () => {
  it('returns name as component/propTypes/defaultProps/export', () => {
    const options = {
      name: 'Header'
    };
    expect(pureComponent(options)).toMatchSnapshot();
  });

  each([
    ['css'],
    ['scss'],
    ['sass'],
    ['less']
  ]).it('returns import with correct suffix: %s when approach is imported', stylesSuffix => {
    const options = {
      name: 'Header',
      stylesApproach: 'imported',
      stylesSuffix
    };
    expect(pureComponent(options)).toMatchSnapshot();
  });

  it('returns radium import when styling is radium', () => {
    const options = {
      name: 'Header',
      stylesApproach: 'radium'
    };
    expect(pureComponent(options)).toMatchSnapshot();
  });

  each([
    ['css'],
    ['scss'],
    ['sass'],
    ['less']
  ]).it('returns styles import with correct suffix: %s when approach is css-modules', stylesSuffix => {
    const options = {
      name: 'Header',
      stylesApproach: 'css-modules',
      stylesSuffix
    };
    expect(pureComponent(options)).toMatchSnapshot();
  });

  it('returns styles from javascript import when approach is styled components', () => {
    const options = {
      name: 'Header',
      stylesApproach: 'styled-components'
    };
    expect(pureComponent(options)).toMatchSnapshot();
  });
});
