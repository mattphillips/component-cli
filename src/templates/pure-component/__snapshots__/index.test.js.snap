// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pure Component returns import with correct suffix: css when approach is imported 1`] = `
"import React from 'react';
import PropTypes from 'prop-types';
import 'styles.css';

const Header = props => {
  return (

  );
};

Header.propTypes = {
};

Header.defaultProps = {
};

export default Header;
"
`;

exports[`Pure Component returns import with correct suffix: less when approach is imported 1`] = `
"import React from 'react';
import PropTypes from 'prop-types';
import 'styles.less';

const Header = props => {
  return (

  );
};

Header.propTypes = {
};

Header.defaultProps = {
};

export default Header;
"
`;

exports[`Pure Component returns import with correct suffix: sass when approach is imported 1`] = `
"import React from 'react';
import PropTypes from 'prop-types';
import 'styles.sass';

const Header = props => {
  return (

  );
};

Header.propTypes = {
};

Header.defaultProps = {
};

export default Header;
"
`;

exports[`Pure Component returns import with correct suffix: scss when approach is imported 1`] = `
"import React from 'react';
import PropTypes from 'prop-types';
import 'styles.scss';

const Header = props => {
  return (

  );
};

Header.propTypes = {
};

Header.defaultProps = {
};

export default Header;
"
`;

exports[`Pure Component returns name as component/propTypes/defaultProps/export 1`] = `
"import React from 'react';
import PropTypes from 'prop-types';
undefined

const Header = props => {
  return (

  );
};

Header.propTypes = {
};

Header.defaultProps = {
};

export default Header;
"
`;

exports[`Pure Component returns radium import when styling is radium 1`] = `
"import React from 'react';
import PropTypes from 'prop-types';
import radium from 'radium';

const Header = props => {
  return (

  );
};

Header.propTypes = {
};

Header.defaultProps = {
};

export default Header;
"
`;

exports[`Pure Component returns styles from javascript import when approach is styled components 1`] = `
"import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles';

const Header = props => {
  return (

  );
};

Header.propTypes = {
};

Header.defaultProps = {
};

export default Header;
"
`;

exports[`Pure Component returns styles import with correct suffix: css when approach is css-modules 1`] = `
"import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.css';

const Header = props => {
  return (

  );
};

Header.propTypes = {
};

Header.defaultProps = {
};

export default Header;
"
`;

exports[`Pure Component returns styles import with correct suffix: less when approach is css-modules 1`] = `
"import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.less';

const Header = props => {
  return (

  );
};

Header.propTypes = {
};

Header.defaultProps = {
};

export default Header;
"
`;

exports[`Pure Component returns styles import with correct suffix: sass when approach is css-modules 1`] = `
"import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.sass';

const Header = props => {
  return (

  );
};

Header.propTypes = {
};

Header.defaultProps = {
};

export default Header;
"
`;

exports[`Pure Component returns styles import with correct suffix: scss when approach is css-modules 1`] = `
"import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scss';

const Header = props => {
  return (

  );
};

Header.propTypes = {
};

Header.defaultProps = {
};

export default Header;
"
`;
